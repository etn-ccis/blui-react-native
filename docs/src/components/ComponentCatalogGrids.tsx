import React from 'react';
import { Grid } from '@mui/material';
import { componentCatalogLinks } from '../__configuration__/ComponentCatalogLinks';
import { useNavigate } from 'react-router-dom';
import { InfoCard } from './InfoCard';

export const ComponentCatalogGrids: React.FC = () => {
    const navigate = useNavigate();

    return (
        <Grid container spacing={6} sx={{ mt: 4 }}>
            {componentCatalogLinks.map((link) => (
                <Grid item xs={12} sm={6} md={4} key={link.title}>
                    <InfoCard
                        spacing={6}
                        source={link.image}
                        aspectRatio={'3x2'}
                        title={link.title}
                        description={''}
                        onClick={(): void => {
                            navigate(`/components/${link.path}`);
                        }}
                    />
                </Grid>
            ))}
        </Grid>
    );
};
