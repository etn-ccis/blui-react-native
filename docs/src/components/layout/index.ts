export * from './SharedAppBar';
export * from './NavigationDrawer';
