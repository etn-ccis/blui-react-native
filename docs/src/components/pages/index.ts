export * from './HomePage';
export * from './MarkdownPage';
export * from './componentPreviewPage';
