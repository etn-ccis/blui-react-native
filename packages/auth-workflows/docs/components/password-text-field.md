# PasswordTextField

Component that renders a text field with a visibility toggle.

<img width="400" alt="passwordTextField" src="../../media/password-text-field.png">

## Usage

```tsx
import { PasswordTextField } from '@brightlayer-ui/react-native-auth-workflow';

...

<PasswordTextField/>
```

## API

See React Native Paper's [TextInputProps](https://callstack.github.io/react-native-paper/docs/components/TextInput/).
