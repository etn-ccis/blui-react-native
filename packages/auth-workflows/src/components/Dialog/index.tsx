export * from './BasicDialog';
export * from './types';
