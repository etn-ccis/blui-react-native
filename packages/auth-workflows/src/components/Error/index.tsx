export * from './ErrorMessageBox';
export * from './ErrorManager';
export * from './types';
