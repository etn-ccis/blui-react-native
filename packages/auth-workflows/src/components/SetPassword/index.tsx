export * from './SetPassword';
export * from './PasswordTextField';
export * from './types';
export * from './PasswordRequirements';
