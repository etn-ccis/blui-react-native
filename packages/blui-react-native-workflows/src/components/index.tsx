export * from './WorkflowCard';
export * from './SetPassword';
export * from './Error';
export * from './Dialog';
export * from './RegistrationWorkflow';
export * from './Spinner';
