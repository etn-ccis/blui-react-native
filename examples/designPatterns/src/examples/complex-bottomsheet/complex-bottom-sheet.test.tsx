import React from 'react';
import renderer from 'react-test-renderer';
import {ComplexBottomSheetScreen} from './components/BottomSheet';
import {sortEvents, filterEvents} from './ComplexBottomSheet';

jest.mock('@react-navigation/native', () => ({
  useNavigation: (): any => ({openDrawer: jest.fn(() => true)}),
}));

describe('Complex Bottom Sheet Tests', () => {
  it('renders the screen', () => {
    const tree = renderer.create(<ComplexBottomSheetScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should sort by time correctly', () => {
    const data = [
      {
        type: 'alarm',
        date: 0,
        active: false,
        location: '',
        device: '',
        details: '',
      },
      {
        type: 'alarm',
        date: 10,
        active: true,
        location: '',
        device: '',
        details: '',
      },
      {
        type: 'settings',
        date: 4,
        active: true,
        location: '',
        device: '',
        details: '',
      },
      {
        type: 'session',
        date: 8,
        active: true,
        location: '',
        device: '',
        details: '',
      },
    ];
    const sortedTime = [
      {
        type: 'alarm',
        date: 10,
        active: true,
        location: '',
        device: '',
        details: '',
      },
      {
        type: 'session',
        date: 8,
        active: true,
        location: '',
        device: '',
        details: '',
      },
      {
        type: 'settings',
        date: 4,
        active: true,
        location: '',
        device: '',
        details: '',
      },
      {
        type: 'alarm',
        date: 0,
        active: false,
        location: '',
        device: '',
        details: '',
      },
    ];

    expect(sortEvents('time', data)).toEqual(sortedTime);
  });

  it('should sort by type correctly', () => {
    const data = [
      {
        type: 'alarm',
        date: 0,
        active: false,
        location: '',
        device: '',
        details: '',
      },
      {
        type: 'alarm',
        date: 10,
        active: true,
        location: '',
        device: '',
        details: '',
      },
      {
        type: 'settings',
        date: 4,
        active: true,
        location: '',
        device: '',
        details: '',
      },
      {
        type: 'session',
        date: 8,
        active: true,
        location: '',
        device: '',
        details: '',
      },
    ];
    const sortedType = [
      {
        type: 'alarm',
        date: 10,
        active: true,
        location: '',
        device: '',
        details: '',
      },
      {
        type: 'alarm',
        date: 0,
        active: false,
        location: '',
        device: '',
        details: '',
      },
      {
        type: 'session',
        date: 8,
        active: true,
        location: '',
        device: '',
        details: '',
      },
      {
        type: 'settings',
        date: 4,
        active: true,
        location: '',
        device: '',
        details: '',
      },
    ];

    expect(sortEvents('type', data)).toEqual(sortedType);
  });

  it('should filter events correctly', () => {
    const data = [
      {
        type: 'alarm',
        date: 0,
        active: false,
        location: '',
        device: '',
        details: '',
      },
      {
        type: 'alarm',
        date: 10,
        active: true,
        location: '',
        device: '',
        details: '',
      },
      {
        type: 'settings',
        date: 4,
        active: true,
        location: '',
        device: '',
        details: '',
      },
      {
        type: 'session',
        date: 8,
        active: true,
        location: '',
        device: '',
        details: '',
      },
    ];

    // show alarms
    const alarms = [
      {
        type: 'alarm',
        date: 0,
        active: false,
        location: '',
        device: '',
        details: '',
      },
      {
        type: 'alarm',
        date: 10,
        active: true,
        location: '',
        device: '',
        details: '',
      },
    ];

    // show sessions
    const sessions = [
      {
        type: 'session',
        date: 8,
        active: true,
        location: '',
        device: '',
        details: '',
      },
    ];

    // show settings and active alarms
    const settingsAndActive = [
      {
        type: 'alarm',
        date: 10,
        active: true,
        location: '',
        device: '',
        details: '',
      },
      {
        type: 'settings',
        date: 4,
        active: true,
        location: '',
        device: '',
        details: '',
      },
    ];

    expect(filterEvents(data, true, true, true, true)).toEqual(data);
    expect(filterEvents(data, true, true, false, false)).toEqual(alarms);
    expect(filterEvents(data, false, false, false, true)).toEqual(sessions);
    expect(filterEvents(data, true, false, true, false)).toEqual(
      settingsAndActive,
    );
  });
});
