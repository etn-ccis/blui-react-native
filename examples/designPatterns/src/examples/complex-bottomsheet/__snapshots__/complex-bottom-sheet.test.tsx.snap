// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Complex Bottom Sheet Tests renders the screen 1`] = `
<Modal
  animationType="none"
  deviceHeight={null}
  deviceWidth={null}
  hardwareAccelerated={false}
  hideModalContentWhileAnimating={false}
  onBackdropPress={[Function]}
  onModalHide={[Function]}
  onModalWillHide={[Function]}
  onModalWillShow={[Function]}
  onRequestClose={[Function]}
  panResponderThreshold={4}
  scrollHorizontal={false}
  scrollOffset={0}
  scrollOffsetMax={0}
  scrollTo={null}
  statusBarTranslucent={false}
  supportedOrientations={
    Array [
      "portrait",
      "landscape",
    ]
  }
  swipeThreshold={100}
  transparent={true}
  visible={false}
>
  <View
    accessible={true}
    focusable={true}
    forwardedRef={[Function]}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "black",
        "bottom": 0,
        "height": 1334,
        "left": 0,
        "opacity": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": 750,
      }
    }
  />
  <View
    deviceHeight={null}
    deviceWidth={null}
    forwardedRef={[Function]}
    hideModalContentWhileAnimating={false}
    onBackdropPress={[Function]}
    onModalHide={[Function]}
    onModalWillHide={[Function]}
    onModalWillShow={[Function]}
    panResponderThreshold={4}
    pointerEvents="box-none"
    scrollHorizontal={false}
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    statusBarTranslucent={false}
    style={
      Object {
        "flex": 1,
        "justifyContent": "center",
        "margin": 37.5,
        "transform": Array [
          Object {
            "translateY": 0,
          },
        ],
      }
    }
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    swipeThreshold={100}
  />
</Modal>
`;
